addons/website_sale/views/product_tag_views.xml

```diff
@@ -7,13 +7,17 @@
         <field name="model">product.tag</field>
         <field name="inherit_id" ref="product.product_tag_form_view"/>
         <field name="arch" type="xml">
-            <xpath expr="//field[@name='color']/.." position="after">
-                <group>
-                    <field name="visible_on_ecommerce"/>
-                </group>
-            </xpath>
-            <field name="color" position="after">
-                <field name="image" widget="image" options="{'size': [0, 55]}"/>
+            <field name="name" position="after">
+                <field name="visible_on_ecommerce"/>
+                <field name="color" widget='color'
+                       attrs="{'invisible': [('visible_on_ecommerce', '!=', True)]}"
+                />
+                <field name="image" widget="image" options="{'size': [0, 55]}"
+                       attrs="{'invisible': [('visible_on_ecommerce', '!=', True)]}"
```

Comment:
Avoid having a mix of multiple-attribute lines and single-attribute lines. when it doesn't fit, put attributes on one line each.